src/shell.js

```diff
@@ -72,6 +72,37 @@ if (!ENVIRONMENT_IS_PTHREAD) PthreadWorkerInit = {};
 var currentScriptUrl = (typeof document !== 'undefined' && document.currentScript) ? document.currentScript.src : undefined;
 #endif
 
+if (typeof Module['locateFile'] !== 'function') {
+  // `/` should be present at the end if `Module['scriptDirectory']` is not empty
+  if (!Module['scriptDirectory']) {
+#if !SUPPORT_BASE64_EMBEDDING
+    if (ENVIRONMENT_IS_NODE) {
+      Module['scriptDirectory'] = __dirname + '/';
+    } else if (ENVIRONMENT_IS_WEB && !document.currentScript.src.startsWith('blob:')) {
+      Module['scriptDirectory'] = document.currentScript.src.split('/').slice(0, -1).join('/') + '/';
+    } else if (ENVIRONMENT_IS_WORKER) {
+      Module['scriptDirectory'] = self.location.href.split('/').slice(0, -1).join('/') + '/';
+    } else {
+      Module['scriptDirectory'] = '';
+    }
+#else
+    Module['scriptDirectory'] = '';
+#endif
+  }
+  if (!Module['memoryInitializerPrefixURL']) {
+    Module['memoryInitializerPrefixURL'] = Module['scriptDirectory'];
+  }
+  if (!Module['pthreadMainPrefixURL']) {
+    Module['pthreadMainPrefixURL'] = Module['scriptDirectory'];
+  }
+  if (!Module['cdInitializerPrefixURL']) {
+    Module['cdInitializerPrefixURL'] = Module['scriptDirectory'];
+  }
+  if (!Module['filePackagePrefixURL']) {
+    Module['filePackagePrefixURL'] = Module['scriptDirectory'];
```

Comment:
Mixing these seems unnecessary, especially if we're working towards removing them in the future - less code lines the better.

Reply:
Similar conditions were present before already, they just were places in various locations, so code size does not increase much. In future we can remove these lines and replace every occurrence of `*PrefixURL` with `scriptDirectory`. I'm open for suggestions on cleaner ways to do this.